tests/functional/api/test_branches.py

```diff
@@ -0,0 +1,17 @@
+"""
+GitLab API:
+https://docs.gitlab.com/ee/api/branches.html
+"""
+
+
+def test_branch_name_with_period(project):
+    # Make sure we can create and get a branch name containing a period '.'
+    branch_name = "my.branch.name"
+    branch = project.branches.create({"branch": branch_name, "ref": "main"})
+    assert branch.name == branch_name
+
+    # Ensure we can get the branch
+    fetched_branch = project.branches.get(branch_name)
+    assert branch.id == fetched_branch.id
+
+    branch.delete()
```

Comment:
I guess this isn't strictly needed as the project will be destroyed anyway and it wouldn't actually get deleted if an assert fails. But I'm ok with this as is :)

Reply:
On the plus side it helps prove that we can `branch.delete()`

Reply:
Ah, that was the point of this :) ok, thanks